python/metatensor-operations/metatensor/operations/reduce_over_samples.py

```diff
@@ -437,12 +428,15 @@ def sum_over_samples_block(
     [[ 4  7 10]
      [17 19 21]]
     """
+    if isinstance(sample_names, str):
+        sample_names = [sample_names]
```

Comment:
This conversion done in the private functions? I think we can skip this here or do we need this due to torchscript?

Reply:
It is not: the private function want `List[str]`, not `Union[List[str], str]`. I can change them to do this though!

Reply:
Adding this to the private functions looks for consistent to me. The public functions are just as wrappers for the private ones setting the parameters to the private ones. Sorry for the extra work.